drivers/source/serial_io.c

```diff
@@ -65,15 +73,24 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
 
-    /* USER CODE END */
+    if (xSemaphoreTake(mutex, UART_MUTEX_BLOCK_TIME) == pdTRUE)
+    {
+        sciSendBytes(text, length);
+        xSemaphoreGive(sciLinMutex);
+        return OBC_ERR_CODE_SUCCESS;
+    }
 }
 
-obc_error_code_t sciPrintf(const char *s, ...){
+return OBC_ERR_CODE_SCI_SEND_FAILED;
```

Comment:
Is this supposed to be the error code I talked about in my other comment? Instead of returning this, it'd be better to return `OBC_ERR_CODE_MUTEX_TIMEOUT` since, if `xSemaphoreTake` fails, it means we timed out when trying to acquire the mutex.